code/modules/clothing/belts/polymorph_belt.dm

```diff
@@ -0,0 +1,152 @@
+	if (.)
+		return
+	if (!isliving(target_mob))
+		return
+	if (ishuman(target_mob) && !ismonkey(target_mob))
+		balloon_alert(user, "target too complex!")
+		return
+	if (!(target_mob.mob_biotypes & MOB_ORGANIC))
+		balloon_alert(user, "organic life only!")
+		return
+	if (isanimal_or_basicmob(target_mob))
+		if (!target_mob.compare_sentience_type(SENTIENCE_ORGANIC))
+			balloon_alert(user, "target too intelligent!")
+			return
+	if (stored_mob_type == target_mob.type)
+		balloon_alert(user, "already scanned!")
+		return
+	if (DOING_INTERACTION_WITH_TARGET(user, target_mob))
+		balloon_alert(user, "busy!")
+		return
+	balloon_alert(user, "scanning...")
+	visible_message(span_notice("[user] begins scanning [target_mob] with [src]."))
+	if (!do_after(user, delay = 5 SECONDS, target = target_mob))
+		return
+	visible_message(span_notice("[user] scans [target_mob] with [src]."))
+	stored_mob_type = target_mob.type
+	update_transform_action()
+	playsound(src, 'sound/machines/ping.ogg', 50, FALSE)
+
+/// Make sure we can transform into the scanned target
+/obj/item/polymorph_belt/proc/update_transform_action()
+	if (isnull(stored_mob_type) || !active)
+		return
+	if (isnull(transform_action))
+		transform_action = add_item_action(/datum/action/cooldown/spell/shapeshift/polymorph_belt)
+	transform_action.update_type(stored_mob_type)
+
+/// Pre-activated polymorph belt
+/obj/item/polymorph_belt/functioning
+	active = TRUE
+	icon_state = "polybelt"
+	worn_icon_state = "polybelt"
+
+/// Ability provided by the polymorph belt
+/datum/action/cooldown/spell/shapeshift/polymorph_belt
+	name = "Invert Polymorphic Field"
+	cooldown_time = 30 SECONDS
+	school = SCHOOL_UNSET
+	invocation_type = INVOCATION_NONE
+	spell_requirements = NONE
+	possible_shapes = list(/mob/living/basic/cockroach)
+	/// Amount of time it takes us to transform back or forth
+	var/channel_time = 3 SECONDS
+
+/datum/action/cooldown/spell/shapeshift/polymorph_belt/before_cast(mob/living/cast_on)
+	. = ..()
+	if (. & SPELL_CANCEL_CAST)
+		return
+	if (channel_time <= 0)
+		return
+	if (DOING_INTERACTION_WITH_TARGET(cast_on, cast_on))
+		return . | SPELL_CANCEL_CAST
+
+	var/old_transform = cast_on.transform
+
+	var/animate_step = channel_time / 6
+	playsound(cast_on, 'sound/effects/wounds/crack1.ogg', 50)
+	animate(cast_on, transform = matrix() * 1.1, time = animate_step, easing = SINE_EASING)
+	animate(transform = matrix() * 0.9, time = animate_step, easing = SINE_EASING)
+	animate(transform = matrix() * 1.2, time = animate_step, easing = SINE_EASING)
+	animate(transform = matrix() * 0.8, time = animate_step, easing = SINE_EASING)
+	animate(transform = matrix() * 1.3, time = animate_step, easing = SINE_EASING)
+	animate(transform = matrix() * 0.1, time = animate_step, easing = SINE_EASING)
+
+	cast_on.balloon_alert(cast_on, "transforming...")
+	if (!do_after(cast_on, delay = channel_time, target = cast_on))
+		animate(cast_on, transform = matrix(), time = 0.25, easing = SINE_EASING)
```

Comment:
```suggestion
		animate(cast_on, transform = matrix(), time = 0.25 SECONDS, easing = SINE_EASING)
```

Did you intend seconds here?

Reply:
Weirdly enough this was a mistake but actually gives the effect I wanted, so I set it to 0 instead.